providers/cloudflare/page_rules.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Terraformer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//cloudflare_access_rule
+//cloudflare_rate_limit
+
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/GoogleCloudPlatform/terraformer/terraform_utils"
+	cf "github.com/cloudflare/cloudflare-go"
+)
+
+type PageRulesGenerator struct {
+	CloudflareService
+}
+
+func (*PageRulesGenerator) createPageRuleResources(api *cf.API, zoneID, zoneName string) ([]terraform_utils.Resource, error) {
+	resources := []terraform_utils.Resource{}
+
+	pageRules, err := api.ListPageRules(zoneID)
+	if err != nil {
+		log.Println(err)
+		return resources, err
+	}
+	for _, pageRule := range pageRules {
+		resources = append(resources, terraform_utils.NewResource(
+			pageRule.ID,
+			fmt.Sprintf("%s_%s", zoneName, pageRule.ID),
+			"cloudflare_page_rule",
+			"cloudflare",
+			map[string]string{
+				"zone_id": zoneID,
+			},
+			[]string{},
+			map[string]interface{}{},
+		))
+	}
+
+	return resources, nil
+
+}
+
+func (g *PageRulesGenerator) InitResources() error {
+	api, err := g.initializeAPI()
+	if err != nil {
+		panic(err)
+		return err
+	}
+
+	zones, err := api.ListZones()
+	if err != nil {
+		panic(err)
```

Comment:
why panic if it's return error?

Reply:
To be honest, I was copying the other services in this provider. I assumed there were reasons beyond my understanding for this madness... :)

I can definitely make it more proper Go, tho. Gimme a day or two.